Description: Fix kfreebsd build failure
Forwarded: https://github.com/brndnmtthws/conky/pull/291
Author: Vincent Cheng <vcheng@debian.org>
Bug-Debian: https://bugs.debian.org/815347
Last-Update: 2016-07-02

--- a/cmake/ConkyPlatformChecks.cmake
+++ b/cmake/ConkyPlatformChecks.cmake
@@ -54,7 +54,7 @@
 
 if(CMAKE_SYSTEM_NAME MATCHES "FreeBSD")
 	set(OS_FREEBSD true)
-	set(conky_libs ${conky_libs} -lkvm -ldevstat)
+	set(conky_libs ${conky_libs} -lkvm -ldevstat -lbsd)
 endif(CMAKE_SYSTEM_NAME MATCHES "FreeBSD")
 
 if(CMAKE_SYSTEM_NAME MATCHES "DragonFly")
--- a/src/freebsd.cc
+++ b/src/freebsd.cc
@@ -725,9 +725,7 @@
 
 	for (i = 0; i < n_processes; i++) {
 		if (!((p[i].ki_flag & P_SYSTEM)) && p[i].ki_comm != NULL) {
-			proc = find_process(p[i].ki_pid);
-			if (!proc)
-				proc = new_process(p[i].ki_pid);
+			proc = get_process(p[i].ki_pid);
 
 			proc->time_stamp = g_time;
 			proc->name = strndup(p[i].ki_comm, text_buffer_size.get(*state));
