Index: conky-1.6.1/src/conky.c
===================================================================
--- conky-1.6.1.orig/src/conky.c	2008-10-10 22:35:12.000000000 +0200
+++ conky-1.6.1/src/conky.c	2008-10-10 22:38:02.000000000 +0200
@@ -4990,14 +4990,14 @@
 #endif /* IMLIB2 */
 
 			OBJ(exec) {
-				read_exec(obj->data.s, p, p_max_size);
+				read_exec(obj->data.s, p, text_buffer_size);
 				remove_deleted_chars(p);
 			}
 			OBJ(execp) {
 				struct information *tmp_info;
 				struct text_object_list *text_objects;
 
-				read_exec(obj->data.s, p, p_max_size);
+				read_exec(obj->data.s, p, text_buffer_size);
 
 				tmp_info = malloc(sizeof(struct information));
 				memcpy(tmp_info, cur, sizeof(struct information));
@@ -5010,7 +5010,7 @@
 			OBJ(execbar) {
 				double barnum;
 
-				read_exec(obj->data.s, p, p_max_size);
+				read_exec(obj->data.s, p, text_buffer_size);
 				barnum = get_barnum(p);
 
 				if (barnum >= 0.0) {
@@ -5023,11 +5023,11 @@
 
 				if(strncasecmp(obj->data.s, LOGGRAPH" ", strlen(LOGGRAPH" ")) == EQUAL) {
 					showaslog = TRUE;
-					read_exec(obj->data.s + strlen(LOGGRAPH" ") * sizeof(char), p, p_max_size);
+					read_exec(obj->data.s + strlen(LOGGRAPH" ") * sizeof(char), p, text_buffer_size);
 				} else if(strncasecmp(obj->data.s, NORMGRAPH" ", strlen(NORMGRAPH" ")) == EQUAL) {
-					read_exec(obj->data.s + strlen(NORMGRAPH" ") * sizeof(char), p, p_max_size);
+					read_exec(obj->data.s + strlen(NORMGRAPH" ") * sizeof(char), p, text_buffer_size);
 				} else {
-					read_exec(obj->data.s, p, p_max_size);
+					read_exec(obj->data.s, p, text_buffer_size);
 				}
 				barnum = get_barnum(p);
 
@@ -5041,7 +5041,7 @@
 						>= obj->data.execi.interval) {
 					double barnum;
 
-					read_exec(obj->data.execi.cmd, p, p_max_size);
+					read_exec(obj->data.execi.cmd, p, text_buffer_size);
 					barnum = get_barnum(p);
 
 					if (barnum >= 0.0) {
@@ -5056,7 +5056,7 @@
 						>= obj->data.execi.interval) {
 					double barnum;
 
-					read_exec(obj->data.execi.cmd, p, p_max_size);
+					read_exec(obj->data.execi.cmd, p, text_buffer_size);
 					barnum = get_barnum(p);
 
 					if (barnum >= 0.0) {
@@ -5071,10 +5071,10 @@
 						>= obj->data.execi.interval
 						&& obj->data.execi.interval != 0) {
 					read_exec(obj->data.execi.cmd, obj->data.execi.buffer,
-						p_max_size);
+						text_buffer_size);
 					obj->data.execi.last_update = current_update_time;
 				}
-				snprintf(p, p_max_size, "%s", obj->data.execi.buffer);
+				snprintf(p, text_buffer_size, "%s", obj->data.execi.buffer);
 			}
 			OBJ(execpi) {
 				struct text_object_list *text_objects = 0;
