Index: conky-1.6.1/src/conky.c
===================================================================
--- conky-1.6.1.orig/src/conky.c	2009-02-16 12:57:46.000000000 +0100
+++ conky-1.6.1/src/conky.c	2009-02-16 13:00:48.000000000 +0100
@@ -5979,11 +5979,11 @@
 							free(timeval);
 							break;
 						case TOP_MEM_RES:
-							human_readable(cur->cpu[obj->data.top.num]->rss,
+							human_readable(cur->memu[obj->data.top.num]->rss,
 									p, 255, "top_rss");
 							break;
 						case TOP_MEM_VSIZE:
-							human_readable(cur->cpu[obj->data.top.num]->vsize,
+							human_readable(cur->memu[obj->data.top.num]->vsize,
 									p, 255, "top_rss");
 							break;
 						default:
